src/crimson/os/seastore/async_cleaner.cc

```diff
@@ -984,6 +984,16 @@ AsyncCleaner::gc_reclaim_space_ret AsyncCleaner::gc_reclaim_space()
       reclaim_state->start_pos,
       reclaim_state->end_pos
     ).si_then([this, &t](auto pin_list) {
+      if (!pin_list.empty()) {
+	auto it = pin_list.begin();
+	auto &first_pin = *it;
+	if (first_pin->get_key() < reclaim_state->start_pos) {
+	  // BackrefManager::get_mappings may include a entry before
+	  // reclaim_state->start_pos, which is semantically inconsistent
+	  // with the requirements of the cleaner
```

Comment:
Does it result in failure without this fix?

If the "if condition" is possible, seems we have skipped a live extent when reclaim a segment, which is incorrect right?
 - The reclaim_state will never go backwards.
 - No extent can be placed across the segment boundary.

Reply:
> Does it result in failure without this fix?

Yes

> If the "if condition" is possible, seems we have skipped a live extent when reclaim a segment, which is incorrect right?

No, that "skipped" live extent has already been rewritten in the previous run of `do_reclaim_space`